src/client/dfuse/pil4dfs/aio.c

```diff
@@ -0,0 +1,502 @@
+/**
+ * (C) Copyright 2024 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#define D_LOGFAC DD_FAC(il)
+
+#include <libaio.h>
+#include <time.h>
+#include <unistd.h>
+#include <pthread.h>
+#include <dlfcn.h>
+#include <sys/syscall.h>
+
+#include <daos.h>
+#include <daos_fs.h>
+#include <daos/debug.h>
+#include <daos/common.h>
+
+#include "pil4dfs_int.h"
+
+/* the number of EQs for aio contexts. (d_aio_eq_count_g + d_eq_count) <= d_eq_count_max */
+uint16_t                 d_aio_eq_count_g;
+
+extern pthread_mutex_t   d_lock_aio_eqs_g;
+extern uint16_t          d_eq_count_max;
+extern uint16_t          d_eq_count;
+extern bool              d_hook_enabled;
+extern struct file_obj  *d_file_list[MAX_OPENED_FILE];
+
+struct d_aio_eq {
+	daos_handle_t    eq;
+	pthread_mutex_t  lock;
+	_Atomic uint64_t n_op_queued;
+	_Atomic uint64_t n_op_done;
+};
+
+/* list of EQs dedicated for aio contexts. */
+static struct d_aio_eq aio_eq_list[MAX_EQ];
```

Comment:
I am concerned with the large numbers of EQs/cart network contexts that are just dedicated just for aio.
It would be better to limit those to 16.

Reply:
ok. I will decrease the limit to 16 in next commit. Thank you!